tree/ntuple/v7/test/CustomStructLinkDef.h

```diff
@@ -66,6 +66,13 @@
 #pragma link C++ class std::map<char, std::map<int, CustomStruct>> +;
 #pragma link C++ class std::map<float, std::map<char, std::int32_t>> +;
 
+#pragma link C++ class std::pair <char, long> +;
```

Comment:
The dictionary requests for `std::pair` do not hurt but they also should not be needed.  What lead you to add them?

Reply:
If I don't include them, the `std::map` tests will fail on builds with runtime modules disabled (see all the failing tests on `ROOT-ubuntu2204/nortcxxmod` in this thread). I'm not really knowledgeable on these modules, so I don't know if this just hides some other underlying issue or not..

Reply:
You can actually see it on the latest Jenkins build failure: https://lcgapp-services.cern.ch/root-jenkins/job/root-pullrequests-build/189474/testReport/projectroot.tree.ntuple.v7/test/gtest_tree_ntuple_v7_test_ntuple_types/, because apparently now it also fails for a `std::set` test that contains a `std::pair` without a dictionary. Which is also weird, because these tests used to pass just fine on the same platform before (it does not seem something changed in the Jenkins/build config since).

Reply:
You will need to reproduce the problem locally and/or instrument:
```
ROOT::Experimental::RPairField::RPairField(std::string_view fieldName,
                                           std::array<std::unique_ptr<Detail::RFieldBase>, 2> &itemFields)
   : ROOT::Experimental::RRecordField(fieldName, std::move(itemFields), {},
                                      "std::pair<" + GetTypeList(itemFields) + ">")
{
   // ISO C++ does not guarantee any specific layout for `std::pair`; query TClass for the member offsets
   fClass = TClass::GetClass(GetType().c_str());
   if (!fClass)
      throw RException(R__FAIL("cannot get type information for " + GetType()));
   fSize = fClass->Size();
   if (fClass->GetDataMember("first") == nullptr || fClass->GetDataMember("second") == nullptr) {
      std::cerr << "The TClass for " << GetType() << " is in state: " << fClass->GetState << " and has: \n";
      fClass->GetListOfDataMembers()->ls();
      fClass->GetStreamerInfo()->ls();
      throw RException(R__FAIL("Insufficient information for " + GetType()));
   }
   fOffsets[0] = fClass->GetDataMember("first")->GetOffset();
   fOffsets[1] = fClass->GetDataMember("second")->GetOffset();
}
```

Reply:
Thanks, I observe the same thing locally. For a `std::map<char,std::int64_t>` field (and thus with a child field of type `std::pair<char,std::int64_t>`) as defined in the test here: https://github.com/root-project/root/blob/b5e909275d6f49d81b1fcbfe02805df457e8de9b/tree/ntuple/v7/test/ntuple_types.cxx#L377 the output of the snippet above is
```
The TClass for std::pair<char,std::int64_t> is in state: 2 and has:
OBJ: TListOfDataMembers	TListOfDataMembers	List of TDataMembers for a class : 0

StreamerInfo for class: pair<char,Long64_t>, checksum=0xb5fb752
  char           first           offset=  0 type= 1 Emulation
  Long64_t       second          offset=  8 type=16 Emulation
   i= 0, first           type=  1, offset=  0, len=1, method=0
   i= 1, second          type= 16, offset=  8, len=1, method=0
unknown file: Failure
C++ exception with description "Insufficient information for std::pair<char,std::int64_t>
```
When I add the dictionary entry for this `std::pair` type the test passes without issue.

Reply:
Let's discuss this in a follow-up issue / PR.

Reply:
> C++ exception with description "Insufficient information for std::pair<char,std::int64_t>

Where does the error comes from (stack trace)?

Reply:
I've opened a issue to further discuss this: https://github.com/root-project/root/issues/14084